### DISCUSSION TITLE
How to Avoid Recompiling `protobuf-src` with System-installed Protobuf in libp2p?

### DISCUSSION THREAD

**Discussion by vinay10949** [libp2p/rust-libp2p#5746] [General]
Hi everyone,

I'm working on  project that includes `libp2p` (version 0.54.1) in a workspace. I’ve noticed that during the build process, `protobuf-src` is being compiled from source, which significantly increases the build time. This is happening despite having Protobuf installed system-wide and setting the `PROTOC` environment variable to point to the system's `protoc` binary.

### Setup Details:
- **Rust version**: 1.71
- **libp2p version**: 0.54.1
- **OS**: Ubuntu 22.04
- **Installed Protobuf**:
  - `sudo apt install protobuf-compiler libprotobuf-dev`
  - `protoc --version` outputs `libprotoc 3.21.9`

### Configuration:
I have set the `PROTOC` environment variable:
```bash
export PROTOC=/usr/bin/protoc
```

### Problem:
Despite having Protobuf installed system-wide, the `protobuf-src` crate still compiles from source during the `cargo build` process. This has been significantly increasing my build time. I’ve checked the libp2p documentation but couldn’t find any specific flags or features that disable Protobuf-related functionality.

### What I’ve Tried:
- **Setting the `PROTOC` environment variable**: This doesn't stop `protobuf-src` from being compiled.
- **Disabling Protobuf in `libp2p` features**: There are no clear flags to disable Protobuf usage in `libp2p` that I could find.
- **Precompiling `.proto` files**: I considered precompiling `.proto` files and committing the generated Rust code, but I’m unsure how to integrate this cleanly with `libp2p`.

### Question:
Has anyone encountered this issue or have suggestions on how to force `libp2p` to use the system-installed Protobuf binaries, or avoid compiling `protobuf-src` entirely? If `libp2p` or one of its dependencies (like `prost-build`) is causing this, is there a workaround?

### Impact:
This is especially problematic in CI environments, where build times are critical. Any tips to reduce build time would be greatly appreciated!

**Comment by drHuangMHT** [comment]
I don't see `protobuf-src` or `protobuf-sys` in our dependency tree(via `cargo tree -i <package name>`). And we don't use `prost` or `prost-build` in most packages we publish, except an example where we showcase how to collect metrics. Can you check your dependency tree using `cargo tree` to see the dependents of `protobuf-src`(you will need `-i` flag to invert the tree)?
